### ISSUE TITLE
Kernel crashes when importing tensorflow version 1.14

### ISSUE THREAD

**Issue by unary-code** [dso-org/deep-symbolic-optimization#70]
My understanding is that the deep-symbolic-optimization repo requires that you have tensorflow version 1.14, and the code in the deep-symbolic-optimization repo does not work if you use any version of tensorflow 2.
I know this because when I used tensorflow version 2.something (I forgot the exact version but I think it was the latest version of the tensorflow) on Google Colab and I executed "pip install -e ./dso", I got some errors like tensorflow's "set_random_seed" method does not exist. Which probably happened because tensorflow 1.14 has "set_random_seed" method but tensorflow 2.something (the tensorflow version I was using) does not have a "set_random_seed" method but tensorflow 1.14 does have a "set_random_seed" method.

By the way, which place in the directory are we supposed to execute "pip install -e ./dso" from? When the current working directory is inside the outer-most "deep-symbolic-optimization" folder, right (This is where I have executing "pip install -e ./dso" from).
Not when the current working directory is inside the outer-most "dso" folder (the outer-most "dso" folder itself contains a "dso" folder, so "./dso" would be a valid path, which is why I am confused).

I tried Google Colab initially, but I got that issue about "set_random_seed" method of tensorflow not found, and Google Colab only allows tensorflow 2.something now, so I gave up on Google Colab.
Then I tried out Jupyter Notebook, which led to the issue of my kernel crashing when I try to import tensorflow.

Steps that I did that led to my issue:
First, on the terminal on my Macbook, I made a virtual environment (using Anaconda) and installed tensorflow (version 1.14) onto that virtual environment.
Then, I opened up that virtual environment on the terminal and then from the terminal, I ran "jupyter notebook".
Then, on the localhost website (Jupyter Notebook's interface is a localhost website) which shows the local files, I created a ipynb file.
Then, in this ipynb file, I did "!pip install tensorflow==1.14", which worked.
Then, I did "import tensorflow as tf" which caused the kernel to crash. I think because I am trying to import tensorflow version 1.14, and perhaps that version is somehow not compatible with some other software (like jupyter notebook).

**Comment by brendenpetersen** [comment]
Yes, DSO currently only works up to TF1.14, but we will add support for Pytorch soon. We know this is a current limitation as many systems don't work on TF1 anymore.

`pip install -e ./dso` should be run from above the outermost `dso` directory. Just inside `deep-symbolic-optimization`.

Lastly, I'd suggest installing in a terminal, not in a jupyter notebook. I have also only tested with pip, not Anaconda, following the `README`. After installing that way, notebooks on top should work fine.

**Comment by unary-code** [comment]
Hey, thanks for the suggestions and advice.
When I was on the terminal on an Anaconda virtual environment and I ran "python -m dso.run ../BasicExampleConfigFile.json", I got this error:
"/opt/anaconda3/envs/dso_env/bin/python: No module named dso.run"

**Comment by unary-code** [comment]
Meanwhile for when I did the exact commands as the Github README, I got the same error as when I used Anaconda to create a virtual environment (see below):

When I was on the terminal on a Python virtual environment ("python3 -m venv venv3 " followed by "source venv3/bin/activate ") and I was inside the "deep-symbolic-optimization" folder, and I ran "pip install -e ./dso", I got this error message. Any idea how to fix this error message or if this error message is the reason why I do not have a module named "dso.run"?

"distutils.errors.DistutilsError: Command '['/Users/[USERNAME]/Documents/[Folder Name]/deep-symbolic-optimization/venv3/bin/python', '-m', 'pip', '--disable-pip-version-check', 'wheel', '--no-deps', '-w', '/var/folders/hk/7vbk1d0x74dcjw_wpf8mp38h0000gn/T/tmp6dg6__h1', '--quiet', 'numpy']' returned non-zero exit status 1.
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error

× Getting requirements to build editable did not run successfully.
│ exit code: 1
╰─> See above for output.

note: This error originates from a subprocess, and is likely not a problem with pip.
"
